chia/wallet/vault/vault_info.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from chia_rs import G1Element
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint64
+from chia.util.streamable import Streamable, streamable
+from chia.wallet.lineage_proof import LineageProof
+
+
+@streamable
+@dataclass(frozen=True)
+class VaultInfo(Streamable):
+    coin: Coin
+    launcher_id: bytes32
```

Comment:
Our whole wallet service is parametrized by this value when running in vault mode. I don't think this property is necessary.

Reply:
I removed the launcher_coin_id field. It was a bit confusing because we'd termed the pk as the launcher early on in development. Have since removed this, so what was launcher_coin_id is now launcher_id. Perhaps it should have a different name though for clarity?

Reply:
I'm still a bit confused, can anything in the vault wallet that does `self.vault_info.launcher_id` not equally do `self.wallet_state_manager.observation_root`?

Reply:
yeah, it's confusing... `vault_info.launcher_id` refers to the vault singleton's launcher coin id. But `vault_root.launcher_id` refers to the vault's pubkey. We need `vault_info.launcher_id` in a few places, and can use the observation root elsewhere. So might be worth renaming `vault_info.launcher_id` to `singleton_launcher_id` or something?

Reply:
Oh this is where I'm misunderstanding, I thought the observation root was the vault singleton ID, that's what I was expecting it to be.  We get the pubkey from the singleton after we have synced it from the users initial piece of information which is the launcher ID.

Reply:
oh yeah you're right, not sure what I was thinking there. `vault_root.launcher_id` is the launcher coin ID. I noticed a mistake in the test setup that it was taking the eve coin as the launcher, so fixed that. And added a property for launcher_id to get pulled from the observation root. Sorry for the confusion :)